src/main/java/omero/cmd/graphs/DuplicateI.java

```diff
@@ -137,13 +179,98 @@ public DuplicateI(ACLVoter aclVoter, Roles securityRoles, GraphPathBean graphPat
         this.targetClasses = targetClasses;
         this.graphPolicy = graphPolicy;
         this.unnullable = unnullable;
+        this.managedRepositoryRewrites = new PrefixSubstituter<FsFile>(
+                (prefix, file) -> file.getPathFrom(prefix) != null,
+                (prefix, file) -> FsFile.concatenate(prefix, file),
+                (prefix, file) -> file.getPathFrom(prefix));
+        this.managedRepository = applicationContext.getBean(ManagedRepositoryI.class);
+        this.pathServices = ImmutableMap.of(
+                OriginalFile.class, applicationContext.getBean("/OMERO/Files", OriginalFilesService.class)::getFilesPath,
+                Pixels.class, applicationContext.getBean("/OMERO/Pixels", PixelsService.class)::getPixelsPath,
+                Thumbnail.class, applicationContext.getBean("/OMERO/Thumbs", ThumbnailService.class)::getThumbnailPath);
     }
 
     @Override
     public Map<String, String> getCallContext() {
         return null;
     }
 
+    /**
+     * Create a duplicate of a file on the filesystem.
+     * @param from the file to duplicate
+     * @param to where to create the duplicate
+     * @throws IOException if the file duplication failed
+     */
+    private void duplicateFile(final Path from, final Path to) throws IOException {
+        final boolean isFromManaged = isPathManaged.test(from);
+        final boolean isToManaged = isPathManaged.test(to);
+        if (isFromManaged && isToManaged) {
+            try {
+                /* safe to create a hard link in managed repository */
+                Files.createLink(to, from);
+                LOGGER.debug("linked {} to {}", from, to);
+            } catch (FileSystemException fse) {
+                LOGGER.debug("failed to link {} to {}", from, to, fse);
```

Comment:
This should perhaps be at a higher level. Could it point to some form of misconfiguration?

Additionally, I still see an RFE for disallowing non-symlinks. The necessary rollback is in place in case this block threw an exception, right?

Reply:
"Limit increase in disk usage" could definitely be a followup PR, though with maybe a subclass of the request type to non-breakingly specify that limit. Rollback should be good but I definitely encouraging testing that!

Reply:
(I could create an issue elaborating a little on the further work I suggest in the PR description.)

Reply:
To your other question, I'd expect this case to be most likely to occur for managed repositories that are split across multiple partitions, that's why I left it at DEBUG.

Reply:
File copies in the managed repository are now quantified by https://github.com/mtbc/omero-blitz/commit/e2eb9bea at INFO level: "file bytes copied ... MANAGED ..."